packages/adblocker-webextension/src/index.ts

```diff
@@ -128,19 +139,98 @@ export function updateResponseHeadersWithCSP(
   return { responseHeaders };
 }
 
+const HTML_FILTERABLE_MIME_TYPES = new Set([
+  'application/javascript',
+  'application/json',
+  'application/mpegurl',
+  'application/vnd.api+json',
+  'application/vnd.apple.mpegurl',
+  'application/vnd.apple.mpegurl.audio',
+  'application/x-javascript',
+  'application/x-mpegurl',
+  'application/xhtml+xml',
+  'application/xml',
+  'audio/mpegurl',
+  'audio/x-mpegurl',
+]);
+export const MAXIMUM_RESPONSE_BUFFER_SIZE = 10 * 1024 * 1024;
+
+function getHeaderFromDetails(
+  details: OnHeadersReceivedDetailsType,
+  headerName: WebRequest.HttpHeadersItemType['name'],
+): string | undefined {
+  return details.responseHeaders?.find((header) => header.name.toLowerCase() === headerName)
+    ?.value;
+}
+
+export function shouldApplyReplaceSelectors(
+  request: Request<OnBeforeRequestDetailsType | OnHeadersReceivedDetailsType>,
+): boolean {
+  const details = request._originalRequestDetails as OnHeadersReceivedDetailsType;
+
+  // In case of undefined error of xhr/fetch and any kind of network activities
+  if (details.statusCode === 0) {
+    return false;
+  }
+
+  // If status code is non 2xx or 4xx, 5xx
+  const statusCodeScope = (details.statusCode / 100) >>> 0;
+  if (statusCodeScope !== 2 && statusCodeScope < 4) {
+    return false;
+  }
+
+  if (getHeaderFromDetails(details, 'content-disposition')?.startsWith('inline') === false) {
+    return false;
+  }
+
+  if (
+    parseInt(getHeaderFromDetails(details, 'content-length') ?? '0', 10) >=
+    MAXIMUM_RESPONSE_BUFFER_SIZE
+  ) {
+    return false;
+  }
+
+  const contentTypeHeader = getHeaderFromDetails(details, 'content-type');
+  if (
+    contentTypeHeader !== undefined &&
+    (contentTypeHeader?.startsWith('text') || HTML_FILTERABLE_MIME_TYPES.has(contentTypeHeader))
+  ) {
+    return true;
+  }
+
+  if (
+    !request.type.startsWith('image') &&
+    request.type !== 'font' &&
+    request.type !== 'websocket' &&
+    request.type !== 'media'
+  ) {
+    return true;
+  }
+
+  return false;
```

Comment:
Would it make sense to instead include a list of types for which we do want to perform filtering? I think it'd be clearer.

Reply:
Leaving ref [`28d3fcd` (#14)](https://github.com/seia-soto/adblocker/pull/14/commits/28d3fcd26deb51d87b113a7d1f5645fc26023364)